File: 02_Algoritmos/Exercicio_7.java
import java.text.DecimalFormat;
import java.util.Scanner;

public class Exercicio_7 {
    private double saldo;

    public Exercicio_7() {
        saldo = 0.00;
    }

    public double getSaldo() {
        return saldo;
    }

    public void depositar(double valor) {
        if (valor > 0) {
            saldo += valor;
            System.out.println("Depósito de " + formatarValor(valor) + " efetuado.");
        } else {
            System.out.println("Valor de depósito inválido. Digite um valor positivo.");
        }
    }

    public void sacar(double valor) {
        if (valor > 0 && valor <= saldo) {
            saldo -= valor;
            System.out.println("Saque de " + formatarValor(valor) + " efetuado.");
        } else {
            System.out.println("Saque não permitido, saldo insuficiente.");
        }
    }

    private String formatarValor(double valor) {
        DecimalFormat formato = new DecimalFormat("#0.00");
        return formato.format(valor);
    }
    public static void main(String[] args) {
        Exercicio_7 conta = new Exercicio_7();
        Scanner scanner = new Scanner(System.in);

        while (true) {
            System.out.println("Escolha uma opção:");
            System.out.println("1. Ver saldo");
            System.out.println("2. Depositar");
            System.out.println("3. Sacar");
            System.out.println("4. Sair");
            System.out.print("Opção: ");

            int escolha = scanner.nextInt();

            switch (escolha) {
                case 1:
                    System.out.println("Saldo é: " + conta.getSaldo());
                    break;
                case 2:
                    System.out.print("Digite o valor a depositar: ");
                    double valorDeposito = scanner.nextDouble();
                    conta.depositar(valorDeposito);
                    break;
                case 3:
                    System.out.print("Digite o valor a retirar: ");
                    double valorSaque = scanner.nextDouble();
                    conta.sacar(valorSaque);
                    break;
                case 4:
                    System.out.println("Encerrando o programa.");
                    scanner.close();
                    System.exit(0);
                default:
                    System.out.println("Opção inválida. Por favor, escolha uma opção válida.");
            }
        }
    }
}

File: Algoritmos/Exercicio_7.py
class ContaBancaria:
    def __init__(self):
        self.saldo = 0.00

    def get_saldo(self):
        return self.saldo

    def depositar(self, valor):
        if valor > 0:
            self.saldo += valor
            print(f"Depósito de {self.formatar_valor(valor)} efetuado.")
        else:
            print("Valor de depósito inválido. Digite um valor positivo.")

    def sacar(self, valor):
        if valor > 0 and valor <= self.saldo:
            self.saldo -= valor
            print(f"Saque de {self.formatar_valor(valor)} efetuado.")
        else:
            print("Saque não permitido, saldo insuficiente.")

    def formatar_valor(self, valor):
        return "{:.2f}".format(valor)

if __name__ == "__main__":
    conta = ContaBancaria()

    while True:
        print("Escolha uma opção:")
        print("1. Ver saldo")
        print("2. Depositar")
        print("3. Sacar")
        print("4. Sair")

        opcao = input("Opção: ")

        if opcao == "1":
            print(f"Saldo é: {conta.get_saldo()}")
        elif opcao == "2":
            valor_deposito = float(input("Digite o valor a depositar: "))
            conta.depositar(valor_deposito)
        elif opcao == "3":
            valor_saque = float(input("Digite o valor a retirar: "))
            conta.sacar(valor_saque)
        elif opcao == "4":
            print("Encerrando o programa.")
            break
        else:
            print("Opção inválida. Por favor, escolha uma opção válida.")
